forbidigo/forbidigo.go

```diff
@@ -164,24 +169,35 @@ func (v *visitor) Visit(node ast.Node) ast.Visitor {
 		}
 		return v
 	case *ast.SelectorExpr:
+		selectorExpr = node
 	case *ast.Ident:
 	default:
 		return v
 	}
-	for _, p := range v.linter.patterns {
-		if p.pattern.MatchString(v.textFor(node)) && !v.permit(node) {
+
+	srcText := v.textFor(node)
+	v.match(node, srcText, srcText, patternFeatureSrc)
+	if pkgText := v.textWithPackage(selectorExpr); pkgText != "" {
+		v.match(node, pkgText, srcText, patternFeaturePackage)
+	}
+	return nil
+}
+
+func (v *visitor) match(node ast.Node, matchText, srcText string, features patternFeature) {
```

Comment:
I've been playing around with a refactor of this where I inline this code above, given my desire not to partition the patterns by "feature."  It would work by find finding the package name the first time (and caching it) and then deciding whether to use the text with the package name or the text without on a per-pattern basis.  Would you be open to that approach?

Reply:
That sounds better than looking up the package name repeatedly.

Do you want me to make a similar change in this PR or do you want to refactor after merging? I'm fine with both.

Reply:
I'd welcome that change on this PR.

Reply:
Done.